### PR TITLE
build/ubuntu: build apt local repository

### PR DESCRIPTION
**Pull request by jialeif**

### CODE REVIEW

File: build/ubuntu-22.04/build-repo.sh [hunk 1]
```diff
@@ -42,51 +60,67 @@ build_kernel () {
...
+    # so the local file installation keeps same as before.
+    pushd $GUEST_REPO/more
+    mv $GUEST_DEFAULT_PKG ../
+    popd
+
+    pushd $HOST_REPO/more
+    mv $HOST_DEFAULT_PKG ../
+    popd
+
+    if ! command -v "dpkg-scanpackages"
```
Comment (kenplusplus) [inline build/ubuntu-22.04/build-repo.sh:111]: Let check all command before any actions. So others can fix all potential gap before building

File: build/ubuntu-22.04/build-repo.sh [hunk 2]
```diff
@@ -42,51 +60,67 @@ build_kernel () {
...
 }
 
 build_check "$1"
 
 pushd "$THIS_DIR"
-mkdir -p $GUEST_REPO
-mkdir -p $HOST_REPO
-
-set -ex
+mkdir -p $GUEST_REPO/more
```
Comment (kenplusplus) [inline build/ubuntu-22.04/build-repo.sh:122]: why need more but not "pool" or "main" or "amd64"

**Review by kenplusplus — CHANGES_REQUESTED** [review]

File: build/ubuntu-22.04/build-repo.sh (hunk 2 above)
Comment (jialeif) [reply build/ubuntu-22.04/build-repo.sh:122]: Tools like mini-dinstall will generate "pool" or "main" or "amd64".
For public, dpkg-scanpackages can simply the usage like createrepo_c for RHEL.
The packages in root directory are must for TDX full stack, the others are placed in more so the repo provides all packages we build.

File: build/ubuntu-22.04/build-repo.sh [hunk 3]
```diff
@@ -8,6 +10,22 @@ LOG_DIR="${THIS_DIR}/build-logs"
 
 export DEBIAN_FRONTEND=noninteractive
 
+GUEST_DEFAULT_PKG=" \
+shim_*_amd64.deb \
+grub-efi-amd64_*_amd64.deb grub-efi-amd64-bin_*_amd64.deb \
+linux-image-unsigned-6.2.16-*.deb linux-headers-6.2.16-* linux-modules-6.2.16-* \
```
[the remark this answers is not in the document]
Comment (jialeif) [reply build/ubuntu-22.04/build-repo.sh:16]: the version limit can be removed.

File: build/ubuntu-22.04/build-repo.sh (hunk 1 above)
Comment (jialeif) [reply build/ubuntu-22.04/build-repo.sh:111]: Will move to build_check

File: build/ubuntu-22.04/build-repo.sh (hunk 2 above)
Comment (kenplusplus) [reply build/ubuntu-22.04/build-repo.sh:122]: who defines the naming convention "more"? I saw all packages are put into the directory more. I did not understand what meaning of "more"

File: build/ubuntu-22.04/README.md [hunk 4]
```diff
@@ -48,3 +48,12 @@ Download is performed unsandboxed as root as file as file ... couldn't be access
 
 You can also install packages for TDX guest using similar steps after copying the
 whole guest packages into TDX guest VM.
+
+Another way, which supports configuring a local apt repository:
+
+```
+cp host_repo /tmp/ -fr
+cat > /etc/apt/sources.list.d/tdx-host.list << EOL
```
[the remark this answers is not in the document]
Comment (jialeif) [reply build/ubuntu-22.04/README.md]: updated

**Review by kenplusplus — APPROVED** [review]
LGTM

**Pull request merged**